firrtl/src/main/scala/firrtl/ir/IR.scala

```diff
@@ -445,6 +445,23 @@ case class LayerBlock(info: Info, layer: String, body: Statement) extends Statem
 case class DefOption(info: Info, name: String, cases: Seq[DefOptionCase])
 case class DefOptionCase(info: Info, name: String)
 
+case class IntrinsicExpr(
+  intrinsic: String,
```

Comment:
Expressions don't have source locators, e.g., primops don't.  And if they did it is basically dead AFAICT -- there's no syntax for emitting them.  If this exists in the future, probably we'll want to add that to all our expressions too!

The nodes these are assigned to[1] will get the source locator from where `IntrinsicExpr` is called -- which is stored in the `DefIntrinsicExpr` it creates-- and given to the `DefNode` (FIRRTL thing) while the `IntrinsicExpr` itself doesn't contain it.

[1]  These must be spilled (or ensured they appear exactly once) -- because they may have side-effects.

That's my thinking anyway -- does that make sense?

Anyway yes source locators make it to the output! I tested this earlier, if you don't hear back it's still working 😉 .

Reply:
Added a println to one of the IntrinsicSpec tests to confirm hadn't somehow regressed, and LGTM:

```
    node _test_T = intrinsic(test : UInt<32>, f, g) @[src/test/scala/chiselTests/IntrinsicSpec.scala 44:51]
```

Reply:
Thanks for the additional explanation! Okay cool so the source locator is still on the `node`, then LGTM!